projects/com.oracle.truffle.llvm.asm.amd64/src/com/oracle/truffle/llvm/asm/amd64/AsmFactory.java

```diff
@@ -227,37 +745,171 @@ protected void addFrameSlot(String reg, LLVMBaseType type) {
     }
 
     private void getArguments() {
-        for (int i = LLVMCallNode.ARG_START_INDEX; i < registers.size(); i++) {
-            String reg = registers.get(i);
-            if (reg != null) {
-                int index = getRegisterIndex(reg);
-                if (index != 0) {
-                    LLVMI32Node node = LLVMI32ArgNodeGen.create(index);
+        String[] tokens = asmFlags.substring(1, asmFlags.length() - 1).split(",");
+        int index = 1;
+        Set<String> todoRegisters = new HashSet<>(registers);
+        for (String token : tokens) {
+            if (token.charAt(0) == '~') {
+                continue;
+            }
+            int start = token.indexOf('{');
+            int end = token.lastIndexOf('}');
+            // FIXME: start or end could be -1
```

Comment:
could you add an assertion please.
so if that issue is triggered, we get an exception and immediately know where to fix it.